lua/lean/infoview.lua

```diff
@@ -590,6 +590,16 @@ function Info:render()
   self.__renderer:render()
   if self.__diff_pin then self.__diff_renderer:render() end
 
+  -- Set the cursor to the line with first goal (just after the marker).
+  for i, line in ipairs(vim.api.nvim_buf_get_lines(self.__renderer.buf, 0, -1, false)) do
+    if line:find("^⊢ ") then
+      vim.api.nvim_win_call(self.__infoview.window,
+        function() vim.cmd.normal(i .. 'z-2l') end
+      )
+      break
+    end
+  end
```

Comment:
Hmm, it's been a bit so I'd have to review some of our codebase again, but where all is `Info:render()` being called? I have a concern that this will move around the view while you're interacting with the infoview -- it should only happen when you "switch" infoviews (i.e. move somewhere else in the source), not within a single interactive infoview "session.

Reply:
Also, we need to guard this with `if self.__infoview.window then ...`, otherwise we get error messages when the window is closed.

Reply:
Thanks! You seem correct, and I never finished separating out Infoview / removing Info. It's definitely counterintuitive to me that `render` is called on closed infoviews, but it seems it indeed is... Will have a look and add a test.